dev-util/kyua/patches/kyua-0.13.patchset

```diff
@@ -0,0 +1,660 @@
++#else
++            name_max = 256; /* Wild guess */
++#endif
++        const size_t length = offsetof(::dirent, d_name) + name_max + 1;
++        _dirent = static_cast< ::dirent* >(operator new(length));
++
+         // Initialize our first directory entry.  Note that this may actually
+         // close the directory we just opened if the directory happens to be
+         // empty -- but directories are never empty because they at least have
+@@ -174,6 +185,8 @@ struct utils::fs::detail::directory_iterator::impl : utils::noncopyable {
+     {
+         if (_dirp != NULL)
+             close();
++        delete _dirent;
++        _dirent = NULL;
+     }
+ 
+     /// Closes the directory and invalidates the iterator.
+@@ -198,7 +211,7 @@ struct utils::fs::detail::directory_iterator::impl : utils::noncopyable {
+     {
+         ::dirent* result;
+ 
+-        if (::readdir_r(_dirp, &_dirent, &result) == -1) {
++        if (::readdir_r(_dirp, _dirent, &result) == -1) {
+             const int original_errno = errno;
+             throw fs::system_error(F("readdir_r(%s) failed") % _path,
+                                    original_errno);
+@@ -207,7 +220,7 @@ struct utils::fs::detail::directory_iterator::impl : utils::noncopyable {
+             _entry.reset(NULL);
+             close();
+         } else {
+-            _entry.reset(new directory_entry(_dirent.d_name));
++            _entry.reset(new directory_entry(_dirent->d_name));
+         }
+     }
+ };
+-- 
+2.15.0
+
+
+From b3bfe5af715d558cc1da1ad9c478e82c4082d3e5 Mon Sep 17 00:00:00 2001
+From: Leorize <alaviss@users.noreply.github.com>
+Date: Fri, 5 Jan 2018 20:36:11 +0700
+Subject: utils/process/status: use WIFCORED on Haiku
+
+
+diff --git a/utils/process/operations_test.cpp b/utils/process/operations_test.cpp
+index 708a740..7a50bfc 100644
+--- a/utils/process/operations_test.cpp
++++ b/utils/process/operations_test.cpp
+@@ -53,6 +53,9 @@ extern "C" {
+ namespace fs = utils::fs;
+ namespace process = utils::process;
+ 
++#if defined(__HAIKU__)
++#   define WCOREDUMP(x) WIFCORED(x)
++#endif
+ 
+ namespace {
+ 
+diff --git a/utils/process/status.cpp b/utils/process/status.cpp
+index a3cea8e..98605c7 100644
+--- a/utils/process/status.cpp
++++ b/utils/process/status.cpp
+@@ -41,7 +41,9 @@ namespace process = utils::process;
+ using utils::none;
+ using utils::optional;
+ 
+-#if !defined(WCOREDUMP)
++#if defined(__HAIKU__)
++#   define WCOREDUMP(x) WIFCORED(x)
++#elif !defined(WCOREDUMP)
+ #   define WCOREDUMP(x) false
+ #endif
+ 
+diff --git a/utils/process/status_test.cpp b/utils/process/status_test.cpp
+index 5a3e19e..603a90d 100644
+--- a/utils/process/status_test.cpp
++++ b/utils/process/status_test.cpp
+@@ -187,7 +187,7 @@ ATF_TEST_CASE_BODY(integration__coredump)
+     ATF_REQUIRE(!coredump.exited());
+     ATF_REQUIRE(coredump.signaled());
+     ATF_REQUIRE_EQ(SIGQUIT, coredump.termsig());
+-#if !defined(WCOREDUMP)
++#if !defined(WCOREDUMP) || !defined(__HAIKU__)
```

Comment:
I might be wrong, but I think what you want here was a `&&`, not a `||`:
```diff
-+#if !defined(WCOREDUMP) || !defined(__HAIKU__)
++#if !defined(WCOREDUMP) && !defined(__HAIKU__)
```

BTW, it would be cool to include `?x86` in `ARCHITECTURES`. It does not cost much and it can be handy.
After all, that arch is not dead yet ;) Thanks :)

Reply:
It's probably an artifact of patching while sleepy :laughing:, it wasn't even necessary.